kart/dataset_mixins.py

```diff
@@ -46,3 +50,130 @@ def diff_to_wc(self, repo, ds_filter=DatasetKeyFilter.MATCH_ALL):
         """
         # Subclasses to override.
         pass
+
+    def get_raw_diff_for_subtree(self, other, subtree_name, reverse=False):
+        """
+        Get a pygit2.Diff of the diff between some subtree of this dataset, and the same subtree of another dataset
+        (generally the "same" dataset at a different revision).
+        """
+
+        flags = pygit2.GIT_DIFF_SKIP_BINARY_CHECK
+        self_subtree = self.get_subtree(subtree_name)
+        other_subtree = other.get_subtree(subtree_name) if other else self._empty_tree
+
+        diff = self_subtree.diff_to_tree(other_subtree, flags=flags, swap=reverse)
+        self.L.debug(
+            "diff %s (%s -> %s / %s): %s changes",
+            subtree_name,
+            self_subtree.id,
+            other_subtree.id if other_subtree else None,
+            "R" if reverse else "F",
+            len(diff),
+        )
+        return diff
+
+    _INSERT_UPDATE_DELETE = (
+        pygit2.GIT_DELTA_ADDED,
+        pygit2.GIT_DELTA_MODIFIED,
+        pygit2.GIT_DELTA_DELETED,
+    )
+    _INSERT_UPDATE = (pygit2.GIT_DELTA_ADDED, pygit2.GIT_DELTA_MODIFIED)
+    _UPDATE_DELETE = (pygit2.GIT_DELTA_MODIFIED, pygit2.GIT_DELTA_DELETED)
+
+    def diff_subtree(
+        self,
+        other,
+        subtree_name,
+        key_filter=UserStringKeyFilter.MATCH_ALL,
+        key_decoder_method=None,
+        value_decoder_method=None,
```

Comment:
these appear to be _required_ but default to None and aren't checked. I'd suggest not giving them a default value but putting them after a `*` so they're required keyword only args

Reply:
Done